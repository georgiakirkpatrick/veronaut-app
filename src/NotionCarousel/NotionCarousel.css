.NotionCarousel {
    max-height: 80vh;
    overflow: scroll;
}

.NotionCarousel h4 {
    font: italic 600 16px/20px Poppins, sans-serif;
    color: #6d6d6d;
}

.NotionCarousel__items {
    display: flex;
    flex-direction: row;
    width: max-content;
}

.NotionCarousel__notion > div {
    padding-bottom: 10px;
}

.NotionCarousel__notion {
    padding: 3vh 7vw 1vh;
    margin: 3vh 2vw 1vh;
    border: 1px solid #e3e2e2;
    border-radius: 20px;
    width: 80vw;
}

.NotionCarousel__stage {
    text-decoration: underline;
}

@media screen and (min-width: 768px) {
    .NotionCarousel__notion {
        width: 40vw;
        padding: 3vh 4vw 2vh;
        margin: 1vh 1vw;
    }
}

@media screen and (min-width: 1024px) {
    .NotionCarousel__items {
        padding: 2vh 0;
    }
    .NotionCarousel__notion {
        width: 35vw;
        padding: 4vh 3vw;
        margin: 0 2vw 0 0;
    }
}

@media screen and (min-width: 1024px) and (min-height: 1366px) {
    .NotionCarousel__items {
        padding: 1vh 0;
    }

    .NotionCarousel__notion {
        padding: 2vh 3vw;
    }
}

@media screen and (min-width: 1440px) {
    .NotionCarousel__notion {
        width: 25vw;
    }
}

@media screen and (min-width: 2560px) {
    .NotionCarousel h4 {
        font: italic 600 20px/28px Poppins, sans-serif;
        color: #6d6d6d;
    }
    
    .NotionCarousel__notion {
        width: 18vw;
        padding: 2vh 2vw;
    }
}