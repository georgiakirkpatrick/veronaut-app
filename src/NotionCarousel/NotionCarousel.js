import React from 'react'
import countries from '../COUNTRIES'
import './NotionCarousel.css'

const NotionCarousel = props => {
    const {
        certArray,
        notionArray,
        factoryList
    } = props

    const getFactory = factoryId => factoryList.filter(factory => factory.id === factoryId)[0].english_name
    const formatCerts = notion => {
        console.log('notion', notion.certification_ids)
        const notionCerts = notion.certification_ids.map(id => certArray.filter(cert => cert.id === id)[0].text)
        return notionCerts.toString()
    }

    const formattedNotions = notionArray.map(notion => {
        return (
            <section key={notion.id}>
                <h4>{notion.type}</h4>
                <p className="NotionCarousel__stage">Manufacturer</p>
                <p>Country: {countries[notion.manufacturer_country - 1].text}</p>
                <p>Factory: {getFactory(notion.manufacturer_id)}</p>
                {notion.manufacturer_notes ? <p>Manufacturer notes: {notion.manufacturer_notes}</p> : null}

                <p className="NotionCarousel__stage">Material producer</p>
                <p>Country: {countries[notion.material_origin_id - 1].text}</p>
                <p>Factory: {getFactory(notion.material_producer_id)}</p>
                {notion.material_notes ? <p>Producer notes: {notion.manufacturer_id}</p>: null}

                <p className="NotionCarousel__stage">Notion certifications</p>
                <p>{formatCerts(notion)}</p>
            </section>
        )
    })

    return (
        <div 
            className='NotionCarousel'
        >
            <div className='NotionCarousel__items'>
                {formattedNotions}
            </div>
        </div>
    )
}

NotionCarousel.defaultProps = {
    certArray: [],
    notionArray: [],
    factoryList: []
}

export default NotionCarousel