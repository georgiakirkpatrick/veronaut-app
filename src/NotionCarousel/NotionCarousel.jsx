import React from 'react'
import countries from '../COUNTRIES'
import './NotionCarousel.css'

const NotionCarousel = ({
  certArray = [],
  notionArray = [],
  factArray = []
}) => {
  const getFactory = factoryId => factArray.filter(factory => factory.id === factoryId)[0].english_name
  const formatCerts = notion => {
    const notionCerts = notion.certification_ids.map(id => certArray.filter(cert => cert.id === id)[0].text)

    return notionCerts.toString()
  }

  const formattedNotions = notionArray.length > 0
    ? notionArray.map(notion => {
      return (
        <li key={'notion' + notion.id}>
          <section className='NotionCarousel__notion'>
            <h4>{notion.notion_type}</h4>

            <div>
              <p className='NotionCarousel__stage'>Manufacturing</p>
              <p>Manufacturer: {getFactory(notion.manufacturer_id)}</p>
              {notion.manufacturer_notes ? <p>Manufacturer notes: {notion.manufacturer_notes}</p> : null}
              <p>Country: {countries[notion.manufacturer_country - 1].text}</p>
            </div>
              
            <div>
              <p className='NotionCarousel__stage'>Primary material</p>
              <p>Primary material: {notion.material_type}</p>
              <p>Producer: {getFactory(notion.material_producer_id)}</p>
              {notion.material_notes ? <p>Producer notes: {notion.material_notes}</p>: null}
              <p>Country: {countries[notion.material_origin_id - 1].text}</p>
            </div>
              
            {
              notion.certification_ids.length > 0
                ? <div>
                  <p className="NotionCarousel__stage">Notion certifications</p>
                  <p>{formatCerts(notion)}</p>
                </div>
                : <div>
                  <p className="NotionCarousel__stage">Notion certifications</p>
                  <p>None</p>
                </div>
            }
          </section>
        </li>
      )
    })
    : <p>This product does not have any notions</p>

  return (
    <div 
        className='NotionCarousel'
    >
      <ul className='NotionCarousel__items'>
        {formattedNotions}
      </ul>
    </div>
  )
}

export default NotionCarousel