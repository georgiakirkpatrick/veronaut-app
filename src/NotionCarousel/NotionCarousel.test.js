import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import NotionCarousel from './NotionCarousel'

describe('NotionCarousel', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<NotionCarousel />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<NotionCarousel />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})