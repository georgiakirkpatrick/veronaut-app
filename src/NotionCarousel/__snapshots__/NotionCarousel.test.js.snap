// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotionCarousel render the UI as expected 1`] = `
<div
  className="NotionCarousel"
>
  <ul
    className="NotionCarousel__items"
  >
    <p>
      This product does not have any notions
    </p>
  </ul>
</div>
`;
